org.eclipse.lemminx/src/test/java/org/eclipse/lemminx/extensions/contentmodel/AssociateGrammarCodeLensExtensionsTest.java

```diff
@@ -0,0 +1,45 @@
+/*******************************************************************************
+* Copyright (c) 2021 Red Hat Inc. and others.
+* All rights reserved. This program and the accompanying materials
+* which accompanies this distribution, and is available at
+* http://www.eclipse.org/legal/epl-v20.html
+*
+* SPDX-License-Identifier: EPL-2.0
+*
+* Contributors:
+*     Red Hat Inc. - initial API and implementation
+*******************************************************************************/
+package org.eclipse.lemminx.extensions.contentmodel;
+
+import static org.eclipse.lemminx.XMLAssert.cl;
+import static org.eclipse.lemminx.XMLAssert.r;
+import static org.eclipse.lemminx.XMLAssert.testCodeLensFor;
+import static org.eclipse.lemminx.client.ClientCommands.SELECT_FILE;
+
+import org.eclipse.lemminx.commons.BadLocationException;
+import org.junit.jupiter.api.Test;
+
+/**
+ * Associate grammar codelens tests
+ *
+ */
+public class AssociateGrammarCodeLensExtensionsTest {
+
+	@Test
+	public void noGrammar() throws BadLocationException {
+		String xml = "<?xml version=\"1.0\" encoding=\"UTF-8\" ?>\r\n" + //
+				"<foo />";
+		testCodeLensFor(xml, "test.xml", //
+				cl(r(1, 1, 1, 4), "Bind with XSD", SELECT_FILE), //
+				cl(r(1, 1, 1, 4), "Bind with DTD", SELECT_FILE), //
+				cl(r(1, 1, 1, 4), "Bind with xml-model", SELECT_FILE));
+	}
+
+	@Test
+	public void withGrammar() throws BadLocationException {
+		String xml = "<?xml-model href='test.dtd' >\r\n" + //
+				"<foo />";
+		testCodeLensFor(xml, "test.xml");
+	}

```

Comment:
If it's not too difficult, I think it would be nice to add  a test for a client that doesn't support `CodeLensKind.Associate`

Reply:
fixed